### PR TITLE
add esp32-s3-korvo-1 audio and sd card (BSP-446)

### PR DESCRIPTION
**Pull request by pborcin**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

### CODE REVIEW

**Comment by pborcin** [comment]
@tore-espressif @espzav PTAL

File: bsp/esp32_s3_korvo_1/idf_component.yml [hunk 1]
```diff
@@ -15,3 +15,7 @@ dependencies:
   button:
     version: ">=2.5,<4.0"
     public: true
+
+  esp_codec_dev:
+    version: "1.0.3"
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/idf_component.yml:20]: Let's allow automatic bugfix updates
```suggestion
    version: "~1.0.3"
```

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf5.c [hunk 2]
```diff
@@ -8,11 +8,136 @@
...
+    {                                                                                                 \
+        .clk_cfg = I2S_STD_CLK_DEFAULT_CONFIG(_sample_rate),                                          \
+        .slot_cfg = I2S_STD_PHILIP_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_MONO), \
+        .gpio_cfg = BSP_I2S0_GPIO_CFG,                                                                 \
+    }
+
+#define BSP_I2S1_CFG(_sample_rate)                                                                      \
+    {                                                                                                 \
+        .clk_cfg = I2S_TDM_CLK_DEFAULT_CONFIG(_sample_rate),                                          \
+        .slot_cfg = I2S_TDM_PHILIP_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_STEREO, I2S_TDM_SLOT0 | I2S_TDM_SLOT1 | I2S_TDM_SLOT2),\
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf5.c:61]: This TDM configuration is not usual. Could you provide a docs block explaining the settings?

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf4.c [hunk 3]
```diff
@@ -1,16 +1,83 @@
 /*
- * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include "esp_err.h"
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf4.c:7]: This file was probably not updated and will fail to compile.

**Review by tore-espressif — COMMENTED** [review]
Thank you very much for the update @pborcin , I'd like to test on HW before we proceed with this MR.

2 more things:
1. This BSP is not build in any CI job. You should add `sdkconfig.bsp.esp32_s3_korvo_1` file to examples that should work on this BSP. Eg examples/audio
2. Audio for v4.4 will fail to build. Could you please have a look at that too?

**Review by espzav — COMMENTED** [review]
@pborcin Thank you for this PR. Good job! I didn't test it, I haven't got a board. Code LGTM Left only one comment and you should **increase the version of component**.

**Comment by pborcin** [comment]
@tore-espressif @espzav PTAL

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h [hunk 4]
```diff
@@ -62,6 +94,111 @@ typedef enum {
...
 
+/**************************************************************************************************
+ *
+ * I2S audio interface
+ *
+ * There are two devices connected to the I2S peripheral:
+ *  - Codec ES8311 for output (playback) path
+ *  - ADC ES7210 for input (recording) path
+ *
+ * For speaker initialization use bsp_audio_codec_speaker_init() which is inside initialize I2S with bsp_audio_init().
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:105]: Nitpick: Typo?
```suggestion
 * For speaker initialization use bsp_audio_codec_speaker_init() which in turn initializes I2S with bsp_audio_init().
```

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h [hunk 5]
```diff
@@ -13,18 +13,50 @@
...
+#include "esp_codec_dev.h"
 #include "led_indicator.h"
+#include "driver/sdmmc_host.h"
+
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+#include "driver/i2s.h"
+#else
+#include "driver/i2s_std.h"
+#include "driver/i2s_tdm.h"
+#endif
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:26]: Could you also please add capabilities defines? Like this: https://github.com/espressif/esp-bsp/blob/master/bsp/esp-box-3/include/bsp/esp-box-3.h#L30

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h [hunk 6]
```diff
@@ -62,6 +94,111 @@ typedef enum {
...
+esp_err_t bsp_audio_init(const i2s_std_config_t *i2s_config);
+#endif
+
+/**
+ * @brief Get es8311 codec I2S interface (initialized in bsp_audio_init)
+ *
+ * @return
+ *      - Pointer to codec I2S interface handle or NULL when error occured
+ */
+const audio_codec_data_if_t *bsp_audio_get_codec_itf_spk(void);
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:145]: @espzav It seems that when we added `bsp_audio_get_codec_itf` function to BSP APIs, we did not consider the scenarion where 2 I2S peripherals are used for playback and recording. What do you think about this new function?

Comment (tore-espressif) [reply bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:145]: Same applies to `bsp_audio_init` where we pass I2S config only for 1 I2S peripheral 😮‍💨

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h (hunk 5 above)
Comment (tore-espressif) [reply bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:35]: Thanks, could you add _all_ the capabilities macros? And define them as '0' in they are not supported? https://github.com/espressif/esp-bsp/blob/master/bsp/esp32_s3_usb_otg/include/bsp/esp32_s3_usb_otg.h#L25

**Comment by tore-espressif** [comment]
@pborcin I tried on HW and my audio playback is not working (no sound). Is there anything else I need to do to make it work?

**Review by tore-espressif — APPROVED** [review]
LGTM! @espzav should we wait for your review?

**Review by espzav — APPROVED** [review]
LGTM

File: bsp/esp32_s3_korvo_1/idf_component.yml [hunk 7]
```diff
@@ -13,5 +13,9 @@ dependencies:
     public: true
```
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/idf_component.yml:13]: @pborcin Please update version (line 1) to 1.1.0

**Review by tore-espressif — CHANGES_REQUESTED** [review]
@pborcin plese bump the version before merging!

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**